### PR TITLE
Adding unit tests with mock

### PR DESCRIPTION
**Pull request by mdumbrava20**

### CODE REVIEW

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 1]
```diff
@@ -1,31 +1,140 @@
...
+
+    private PaymentRepository paymentRepository;
+
+    private ValidationService validationService;
+
+    private PaymentService paymentService;
 
     @BeforeEach
     void setUp() {
+        userRepository = mock(UserRepository.class);
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Use the @Mock annotation on mocks and @InjectMocks on payment service

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 2]
```diff
@@ -1,31 +1,140 @@
...
+        verify(validationService).validateUserId(userId);
+        verify(validationService).validateAmount(amount);
+        verify(userRepository).findById(userId);
+    }
+
+    @Test
+    void createPaymentWhenUserIsContained() {
+        InMemUserRepository inMemUserRepository = spy(InMemUserRepository.class);
+        paymentService = new PaymentService(inMemUserRepository,
+                paymentRepository, validationService);
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: remove spy and new PaymentService(...)

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 3]
```diff
@@ -1,31 +1,140 @@
...
+        ArgumentCaptor<Payment> paymentCaptor = ArgumentCaptor.forClass(Payment.class);
+
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java:59]: use when (userRepository,findById(1).thenReturn(user);
same for payement repository

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 4]
```diff
@@ -1,31 +1,140 @@
...
+                () -> assertEquals(newMessage, capturedMessage)
+        );
     }
 
     @Test
-    void editMessage() {
+    void getAllByAmountExceedingWithExistingGreaterAmountsInList() {
+        List<Payment> paymentList = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            paymentList.add(new Payment(i, i + 100D, "Payment " + i));
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Move duplicate code to a private method

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 5]
```diff
@@ -1,31 +1,140 @@
...
+        userRepository = mock(UserRepository.class);
+        paymentRepository = mock(PaymentRepository.class);
+        validationService = mock(ValidationService.class);
+        paymentService = new PaymentService(userRepository,
+                paymentRepository, validationService);
+    }
+
+    @Test
+    void createPaymentWhenUserIdNotContained() {
+        Integer userId = 10;
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: use primitives everywhere

File: src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    void validateUserIdWhenParameterIsNull() {
+        assertThrows(IllegalArgumentException.class,
+                () -> basicValidationService.validateUserId(null),
+                "User id must not be null");
+    }
+
+    @Test
+    void validateUserWhenStatusIsInactive() {
+        User user = Mockito.mock(User.class);
+        user.setStatus(Status.INACTIVE);
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java]: remove the setStatus, use Mockito functionality when(user.getStatus)...

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+    private void savingTwoMocksAndOnePayment(Payment payment) {
+        inMemPaymentRepository.save(mock(Payment.class));
+        inMemPaymentRepository.save(mock(Payment.class));
+        inMemPaymentRepository.save(payment);
+    }
+
+    private void savingThreeMocks() {
+        inMemPaymentRepository.save(mock(Payment.class));
+        inMemPaymentRepository.save(mock(Payment.class));
```
Comment (vermurachi) [inline src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: create a separate method that takes an argument as number of payments you want to save and call it here and in previous method like savePaymentMocks(3) or savePaymentMocks(2);

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 1 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Fixed.

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 2 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Fixed.

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 3 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java:59]: Fixed.

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 4 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Fixed.

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 5 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: Fixed.

File: src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java (hunk 6 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/service/BasicValidationServiceTest.java]: Fixed.

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java (hunk 7 above)
Comment (mdumbrava20) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: Fixed.

**Review by vermurachi — APPROVED** [review]

**Pull request merged**